LibAddonMenu-2.0/controls/dropdown.lua

```diff
@@ -1,439 +1,463 @@
+    dropdown.m_dropdown:SetParent(combobox:GetOwningWindow()) -- TODO remove workaround once the problem is fixed in the game
+
+
+    local isMultiSelectionEnabled = GetDefaultValue(dropdownData.multiSelect)
+    control.isMultiSelectionEnabled = isMultiSelectionEnabled
+
+    dropdown.CalculateContentWidth = function()
+        local dataType = ZO_ScrollList_GetDataTypeTable(dropdown.m_scroll, 1)
+
+        local dummy = dataType.pool:AcquireObject()
+        dataType.setupCallback(dummy, {
+            m_owner = dropdown,
+            name = "Dummy"
+        }, dropdown)
+
+        local maxWidth = 0
+        local label = dummy.m_label
+        local entries = dropdown.m_sortedItems
+        local numItems = #entries
+        for index = 1, numItems do
+            label:SetText(entries[index].name)
+            local width = label:GetTextWidth()
+            if (width > maxWidth) then
+                maxWidth = width
+            end
+        end
+
+        dataType.pool:ReleaseObject(dummy.key)
+        return maxWidth
+    end
+
+    dropdown.AdjustDimensions = function()
+        local numItems = #dropdown.m_sortedItems
+        local dropdownObject = dropdown.m_dropdown
+        local scroll = dropdownObject:GetNamedChild("Scroll")
+        local scrollContent = scroll:GetNamedChild("Contents")
+        local anchorOffset = 0
+
+        local contentWidth = dropdown.CalculateContentWidth() + CONTENT_PADDING
+        local visibleRows, min, max = SetDropdownHeight(combobox, dropdown, dropdownData)
+
+        if(numItems > visibleRows) then
+            numItems = visibleRows
+            if isMultiSelectionEnabled then
+                if dropdownData.scrollable ~= nil then
+                    if not scroll.scrollbar:IsHidden() then
+                        contentWidth = contentWidth + SCROLLBAR_PADDING
+                        anchorOffset = -SCROLLBAR_PADDING
+                    else
+                        anchorOffset = -6
+                    end
+                else
+                    anchorOffset = -6
+                end
+            else
+                contentWidth = contentWidth + SCROLLBAR_PADDING
+                anchorOffset = -SCROLLBAR_PADDING
+            end
+        else
+            if isMultiSelectionEnabled then
+                anchorOffset = -6
+            end
+        end
+        local width = zo_max(contentWidth, dropdown.m_container:GetWidth())
+        dropdownObject:SetWidth(width)
+
+        scrollContent:SetAnchor(BOTTOMRIGHT, nil, nil, anchorOffset)
+    end
+
+    --Multiselection
+    if isMultiSelectionEnabled == true then
+        local multiSelectType = GetDefaultValue(dropdownData.multiSelectType)
+        control.data.multiSelectType = multiSelectType or "normal"
+
+        --Add context menu to the multiselect dropdown: Select all / Clear all selections
+        control.CallMultiSelectSetFunc = callMultiSelectSetFunc
+        local mouseUp = combobox:GetHandler("OnMouseUp")

```

Comment:
should not be needed? the SetHandler function allows to set multiple handlers on the same event and the default handler does not interfere with the handler below since it checks for a different button.